### PR TITLE
Add e2e tests for resource processor

### PR DESCRIPTION
**Pull request by asuresh4**
**Description:** Add e2e tests for resource processor.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@asuresh4 this looks good to me. We had a discussion with Bogdan a couple days ago where he suggested that for non-performance related tests we use regular unit tests in the codebase instead of the testbed. I think he has a valid point.
Let's merge this for now but for future we should probably use the testbed for performance, load and other types of tests which actually require that the Collector is run as a separate process.

File: testbed/tests/metric_test.go [hunk 1]
```diff
@@ -79,3 +91,215 @@ func TestMetric10kDPS(t *testing.T) {
...
+
+	mockedConsumedResources := map[string]*otlpmetrics.ResourceMetrics{}
+	err = json.Unmarshal(inputs, &mockedConsumedResources)
+	require.NoError(t, err, "failed to load test dependencies", err)
+	require.Equal(t, 4, len(mockedConsumedResources), "not all mocked resources loaded")
+
+	sender := testbed.NewOTLPMetricDataSender(testbed.GetAvailablePort(t))
+	receiver := testbed.NewOTLPDataReceiver(testbed.GetAvailablePort(t))
+
+	tests := []struct {
```
Comment (tigrannajaryan) [inline testbed/tests/metric_test.go:114]: The only part that I am a bit wary about is that these 4 test cases correspond to 4 json definitions but they are disconnected and are in different files. Can we do this slightly differently to ensure each test case and its corresponding json are near each other? For example have a separate *.go file which contains this data, including the json as a Go string and eliminate the json file.

Comment (asuresh4) [reply testbed/tests/metric_test.go]: Updated to have resource metric definitions as string constants along with the test cases, eliminating the JSON file. Let me know if this seems to be a better approach.

File: testbed/tests/metric_test.go [hunk 2]
```diff
@@ -79,3 +85,69 @@ func TestMetric10kDPS(t *testing.T) {
 	}
 
 }
+
+func TestMetricResourceProcessor(t *testing.T) {
```
Comment (tigrannajaryan) [inline testbed/tests/metric_test.go:89]: Perhaps worth moving this func to resource_processor_test.go too?

**Comment by asuresh4** [comment]
@tigrannajaryan - rebased to rerun the stuck build.

**Pull request merged**